client/network/sync/src/engine.rs

```diff
@@ -0,0 +1,929 @@
+						match result {
+							Ok((id, req)) => self.chain_sync.send_block_request(id, req),
+							Err(BadPeer(id, repu)) => {
+								self.network_service
+									.disconnect_peer(id, self.block_announce_protocol_name.clone());
+								self.network_service.report_peer(id, repu)
+							},
+						}
+					}
+				},
+				ToServiceCommand::JustificationImported(peer, hash, number, success) => {
+					self.chain_sync.on_justification_import(hash, number, success);
+					if !success {
+						log::info!(target: "sync", "💔 Invalid justification provided by {} for #{}", peer, hash);
+						self.network_service
+							.disconnect_peer(peer, self.block_announce_protocol_name.clone());
+						self.network_service.report_peer(
+							peer,
+							sc_peerset::ReputationChange::new_fatal("Invalid justification"),
+						);
+					}
+				},
+				ToServiceCommand::AnnounceBlock(hash, data) => self.announce_block(hash, data),
+				ToServiceCommand::NewBestBlockImported(hash, number) =>
+					self.new_best_block_imported(hash, number),
+				ToServiceCommand::Status(tx) =>
+					if let Err(_) = tx.send(self.chain_sync.status()) {
+						log::warn!(target: "sync", "Failed to respond to `Status` query");
+					},
+				ToServiceCommand::NumActivePeers(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.num_active_peers()) {
+						log::warn!(target: "sync", "response channel closed for `NumActivePeers`");
+					}
+				},
+				ToServiceCommand::SyncState(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.status()) {
+						log::warn!(target: "sync", "response channel closed for `SyncState`");
+					}
+				},
+				ToServiceCommand::BestSeenBlock(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.status().best_seen_block) {
+						log::warn!(target: "sync", "response channel closed for `BestSeenBlock`");
+					}
+				},
+				ToServiceCommand::NumSyncPeers(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.status().num_peers) {
+						log::warn!(target: "sync", "response channel closed for `NumSyncPeers`");
+					}
+				},
+				ToServiceCommand::NumQueuedBlocks(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.status().queued_blocks) {
+						log::warn!(target: "sync", "response channel closed for `NumQueuedBlocks`");
+					}
+				},
+				ToServiceCommand::NumDownloadedBlocks(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.num_downloaded_blocks()) {
+						log::warn!(target: "sync", "response channel closed for `NumDownloadedBlocks`");
+					}
+				},
+				ToServiceCommand::NumSyncRequests(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.num_sync_requests()) {
+						log::warn!(target: "sync", "response channel closed for `NumSyncRequests`");
+					}
+				},
+				ToServiceCommand::PeersInfo(tx) => {
+					let peers_info =
+						self.peers.iter().map(|(id, peer)| (*id, peer.info.clone())).collect();
+					if let Err(_) = tx.send(peers_info) {
+						log::warn!(target: "sync", "response channel closed for `PeersInfo`");
+					}
```

Comment:
We should clearly not log anything of this! When someone throws away the receiver, this is their problem and not ours, especially not that we should print some warning.